lib/embulk/input/marketo/lead.rb

```diff
@@ -91,7 +91,12 @@ def self.generate_columns(metadata)
 
         def init
           @columns = task[:columns]
-          @ranges = task[:ranges][index]
+          if preview?
+            # Try earlier date at first to reduce cache miss hit
+            @ranges = task[:ranges].flatten.sort_by{|range| Time.parse(range["to"])}.reverse
```

Comment:
Got it. I misunderstood 'Try earlier date at first' as 'Try **older** date at first', but the correct behavior would be to try newer date at first, right?

I'd suggest s/earlier/later/ or s/earlier/newer/